.github/workflows/ci.yml

```diff
@@ -25,12 +25,15 @@ jobs:
           - build: linux64
             os: ubuntu-latest
             host_target: x86_64-unknown-linux-gnu
+            env: MIRIFLAGS=-Zmiri-tag-gc=1
```

Comment:
it's weird that this made the linux runner faster, too. It looks like it is actually better to run the GC after every basic block instead of after every 10k blocks.

Reply:
It's not, that was a combination of me doing things wrong and CI timing noise.

This is a mark-and-sweep GC for a resource which sometimes disposes of itself so we benefit doubly from not running the GC too often. There is fixed cost to the mark stage but the amount we sweep away grows if we wait. Plus if an allocation is allocated then deallocated in the time between the GC runs, we never have to interact with it at all.

I did consider trying to hack up some kind of generation system, but between just ignoring small stacks and stacks not modified since the last GC run, the overhead of the sweep part is quite small at the default GC interval.